{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9LmL_bLauer2",
        "outputId": "b29488b2-58dd-43e8-ce60-8b41355e567a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: kafka-python in /usr/local/lib/python3.11/dist-packages (2.0.2)\n",
            "Collecting s3fs\n",
            "  Downloading s3fs-2024.12.0-py3-none-any.whl.metadata (1.6 kB)\n",
            "Collecting aiobotocore<3.0.0,>=2.5.4 (from s3fs)\n",
            "  Downloading aiobotocore-2.18.0-py3-none-any.whl.metadata (23 kB)\n",
            "Collecting fsspec==2024.12.0.* (from s3fs)\n",
            "  Downloading fsspec-2024.12.0-py3-none-any.whl.metadata (11 kB)\n",
            "Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /usr/local/lib/python3.11/dist-packages (from s3fs) (3.11.11)\n",
            "Collecting aioitertools<1.0.0,>=0.5.1 (from aiobotocore<3.0.0,>=2.5.4->s3fs)\n",
            "  Downloading aioitertools-0.12.0-py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting botocore<1.36.2,>=1.36.0 (from aiobotocore<3.0.0,>=2.5.4->s3fs)\n",
            "  Downloading botocore-1.36.1-py3-none-any.whl.metadata (5.7 kB)\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.11/dist-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs) (2.8.2)\n",
            "Collecting jmespath<2.0.0,>=0.7.1 (from aiobotocore<3.0.0,>=2.5.4->s3fs)\n",
            "  Downloading jmespath-1.0.1-py3-none-any.whl.metadata (7.6 kB)\n",
            "Requirement already satisfied: multidict<7.0.0,>=6.0.0 in /usr/local/lib/python3.11/dist-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs) (6.1.0)\n",
            "Requirement already satisfied: urllib3!=2.2.0,<3,>=1.25.4 in /usr/local/lib/python3.11/dist-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs) (2.3.0)\n",
            "Requirement already satisfied: wrapt<2.0.0,>=1.10.10 in /usr/local/lib/python3.11/dist-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs) (1.17.0)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (2.4.4)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (1.3.2)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (24.3.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (1.5.0)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (0.2.1)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (1.18.3)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil<3.0.0,>=2.1->aiobotocore<3.0.0,>=2.5.4->s3fs) (1.17.0)\n",
            "Requirement already satisfied: idna>=2.0 in /usr/local/lib/python3.11/dist-packages (from yarl<2.0,>=1.17.0->aiohttp!=4.0.0a0,!=4.0.0a1->s3fs) (3.10)\n",
            "Downloading s3fs-2024.12.0-py3-none-any.whl (30 kB)\n",
            "Downloading fsspec-2024.12.0-py3-none-any.whl (183 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m183.9/183.9 kB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading aiobotocore-2.18.0-py3-none-any.whl (77 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m77.6/77.6 kB\u001b[0m \u001b[31m7.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading aioitertools-0.12.0-py3-none-any.whl (24 kB)\n",
            "Downloading botocore-1.36.1-py3-none-any.whl (13.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m60.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading jmespath-1.0.1-py3-none-any.whl (20 kB)\n",
            "Installing collected packages: jmespath, fsspec, aioitertools, botocore, aiobotocore, s3fs\n",
            "  Attempting uninstall: fsspec\n",
            "    Found existing installation: fsspec 2024.10.0\n",
            "    Uninstalling fsspec-2024.10.0:\n",
            "      Successfully uninstalled fsspec-2024.10.0\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "gcsfs 2024.10.0 requires fsspec==2024.10.0, but you have fsspec 2024.12.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed aiobotocore-2.18.0 aioitertools-0.12.0 botocore-1.36.1 fsspec-2024.12.0 jmespath-1.0.1 s3fs-2024.12.0\n"
          ]
        }
      ],
      "source": [
        "!pip install kafka-python\n",
        "!pip install s3fs\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "from kafka import KafkaConsumer, KafkaProducer\n",
        "from time import sleep\n",
        "from json import dumps,loads\n",
        "import json"
      ],
      "metadata": {
        "id": "0ASdZ4fRun-B"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "consumer = KafkaConsumer('demo_testing2',\n",
        "                         bootstrap_servers=['16.171.32.64:9092'],\n",
        "                         value_deserializer=lambda x: loads(x.decode('utf-8')))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "O_QYWJeVwuxk",
        "outputId": "8c65aac9-7128-40c2-ed68-6d931f448597"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:kafka.coordinator.consumer:group_id is None: disabling auto-commit.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#for c in consumer:\n",
        "    #print(c.value)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "collapsed": true,
        "id": "KXYL-TFRxPGJ",
        "outputId": "b18cf7b8-fba3-46d2-9b71-47685d1e3208"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat poll expired, leaving group\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat failed for group consumer-group-a because it is rebalancing\n",
            "WARNING:kafka.coordinator:Heartbeat poll expired, leaving group\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'Index': '399001.SZ', 'Date': '2004-03-31', 'Open': 4018.830078, 'High': 4061.669922, 'Low': 4008.0, 'Close': 4060.139893, 'Adj Close': 4060.094482, 'Volume': 22400.0, 'CloseUSD': 649.62238288}\n",
            "{'Index': 'N225', 'Date': '1966-12-21', 'Open': 1404.300049, 'High': 1404.300049, 'Low': 1404.300049, 'Close': 1404.300049, 'Adj Close': 1404.300049, 'Volume': 0.0, 'CloseUSD': 14.04300049}\n",
            "{'Index': 'NYA', 'Date': '1979-04-27', 'Open': 606.929993, 'High': 606.929993, 'Low': 606.929993, 'Close': 606.929993, 'Adj Close': 606.929993, 'Volume': 0.0, 'CloseUSD': 606.929993}\n",
            "{'Index': '000001.SS', 'Date': '2014-11-18', 'Open': 2474.181885, 'High': 2477.052002, 'Low': 2449.812988, 'Close': 2457.529053, 'Adj Close': 2457.529053, 'Volume': 201100.0, 'CloseUSD': 393.20464848000006}\n",
            "{'Index': 'NYA', 'Date': '2013-07-15', 'Open': 9506.459961, 'High': 9531.660156, 'Low': 9496.919922, 'Close': 9520.950195, 'Adj Close': 9520.950195, 'Volume': 2623200000.0, 'CloseUSD': 9520.950195}\n",
            "{'Index': 'GDAXI', 'Date': '1989-09-08', 'Open': 1654.469971, 'High': 1654.469971, 'Low': 1654.469971, 'Close': 1654.469971, 'Adj Close': 1654.469971, 'Volume': 0.0, 'CloseUSD': 2018.45336462}\n",
            "{'Index': 'N100', 'Date': '2002-02-12', 'Open': 765.090027, 'High': 768.340027, 'Low': 755.570007, 'Close': 758.570007, 'Adj Close': 758.570007, 'Volume': 0.0, 'CloseUSD': 925.45540854}\n",
            "{'Index': 'GSPTSE', 'Date': '2017-10-27', 'Open': 15902.90039, 'High': 15963.59961, 'Low': 15872.90039, 'Close': 15953.5, 'Adj Close': 15953.5, 'Volume': 20959390000.0, 'CloseUSD': 13241.405}\n",
            "{'Index': 'GDAXI', 'Date': '1998-07-24', 'Open': 5990.47998, 'High': 6062.029785, 'Low': 5952.819824, 'Close': 6040.580078, 'Adj Close': 6040.580078, 'Volume': 0.0, 'CloseUSD': 7369.507695159999}\n",
            "{'Index': 'SSMI', 'Date': '2017-03-28', 'Open': 8610.19043, 'High': 8629.240234, 'Low': 8568.179688, 'Close': 8597.019531, 'Adj Close': 8597.019531, 'Volume': 51996900.0, 'CloseUSD': 9542.691679410002}\n"
          ]
        },
        {
          "output_type": "error",
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-22-d718415e1172>\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mc\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1191\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_v1\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1192\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1193\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1194\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1195\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mnext_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36mnext_v2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1199\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_message_generator_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1200\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1201\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1202\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1203\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m_message_generator_v2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1114\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_message_generator_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1115\u001b[0m         \u001b[0mtimeout_ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1000\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_consumer_timeout\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1116\u001b[0;31m         \u001b[0mrecord_map\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupdate_offsets\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1117\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mtp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrecords\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miteritems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord_map\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1118\u001b[0m             \u001b[0;31m# Generators are stateful, and it is possible that the tp / records\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36mpoll\u001b[0;34m(self, timeout_ms, max_records, update_offsets)\u001b[0m\n\u001b[1;32m    653\u001b[0m         \u001b[0mremaining\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtimeout_ms\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    654\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 655\u001b[0;31m             \u001b[0mrecords\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_poll_once\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mremaining\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_records\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupdate_offsets\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mupdate_offsets\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    656\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    657\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m_poll_once\u001b[0;34m(self, timeout_ms, max_records, update_offsets)\u001b[0m\n\u001b[1;32m    700\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    701\u001b[0m         \u001b[0mtimeout_ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coordinator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime_to_next_poll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m         \u001b[0;31m# after the long poll, we should check whether the group needs to rebalance\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m         \u001b[0;31m# prior to returning data so that the group can stabilize faster\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/client_async.py\u001b[0m in \u001b[0;36mpoll\u001b[0;34m(self, timeout_ms, future)\u001b[0m\n\u001b[1;32m    600\u001b[0m                     \u001b[0mtimeout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# avoid negative timeouts\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    601\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 602\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_poll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    603\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    604\u001b[0m             \u001b[0;31m# called without the lock to avoid deadlock potential\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/client_async.py\u001b[0m in \u001b[0;36m_poll\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    632\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    633\u001b[0m         \u001b[0mstart_select\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 634\u001b[0;31m         \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    635\u001b[0m         \u001b[0mend_select\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    636\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sensors\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.11/selectors.py\u001b[0m in \u001b[0;36mselect\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    466\u001b[0m             \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    467\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 468\u001b[0;31m                 \u001b[0mfd_event_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_ev\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    469\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mInterruptedError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    470\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Configure AWS CLI\n"
      ],
      "metadata": {
        "id": "sYQxxpUF8Ds1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install awscli"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0dB7xSSc8eTG",
        "outputId": "386bf470-b8a4-4a9e-e7ba-cc3c2a752126"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting awscli\n",
            "  Downloading awscli-1.37.2-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting botocore==1.36.2 (from awscli)\n",
            "  Downloading botocore-1.36.2-py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting docutils<0.17,>=0.10 (from awscli)\n",
            "  Downloading docutils-0.16-py2.py3-none-any.whl.metadata (2.7 kB)\n",
            "Collecting s3transfer<0.12.0,>=0.11.0 (from awscli)\n",
            "  Downloading s3transfer-0.11.1-py3-none-any.whl.metadata (1.7 kB)\n",
            "Requirement already satisfied: PyYAML<6.1,>=3.10 in /usr/local/lib/python3.11/dist-packages (from awscli) (6.0.2)\n",
            "Collecting colorama<0.4.7,>=0.2.5 (from awscli)\n",
            "  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)\n",
            "Collecting rsa<4.8,>=3.1.2 (from awscli)\n",
            "  Downloading rsa-4.7.2-py3-none-any.whl.metadata (3.6 kB)\n",
            "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.11/dist-packages (from botocore==1.36.2->awscli) (1.0.1)\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.11/dist-packages (from botocore==1.36.2->awscli) (2.8.2)\n",
            "Requirement already satisfied: urllib3!=2.2.0,<3,>=1.25.4 in /usr/local/lib/python3.11/dist-packages (from botocore==1.36.2->awscli) (2.3.0)\n",
            "Requirement already satisfied: pyasn1>=0.1.3 in /usr/local/lib/python3.11/dist-packages (from rsa<4.8,>=3.1.2->awscli) (0.6.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil<3.0.0,>=2.1->botocore==1.36.2->awscli) (1.17.0)\n",
            "Downloading awscli-1.37.2-py3-none-any.whl (4.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.6/4.6 MB\u001b[0m \u001b[31m58.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading botocore-1.36.2-py3-none-any.whl (13.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m88.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)\n",
            "Downloading docutils-0.16-py2.py3-none-any.whl (548 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m548.2/548.2 kB\u001b[0m \u001b[31m42.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading rsa-4.7.2-py3-none-any.whl (34 kB)\n",
            "Downloading s3transfer-0.11.1-py3-none-any.whl (84 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m84.1/84.1 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: rsa, docutils, colorama, botocore, s3transfer, awscli\n",
            "  Attempting uninstall: rsa\n",
            "    Found existing installation: rsa 4.9\n",
            "    Uninstalling rsa-4.9:\n",
            "      Successfully uninstalled rsa-4.9\n",
            "  Attempting uninstall: docutils\n",
            "    Found existing installation: docutils 0.21.2\n",
            "    Uninstalling docutils-0.21.2:\n",
            "      Successfully uninstalled docutils-0.21.2\n",
            "  Attempting uninstall: botocore\n",
            "    Found existing installation: botocore 1.36.1\n",
            "    Uninstalling botocore-1.36.1:\n",
            "      Successfully uninstalled botocore-1.36.1\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "aiobotocore 2.18.0 requires botocore<1.36.2,>=1.36.0, but you have botocore 1.36.2 which is incompatible.\n",
            "sphinx 8.1.3 requires docutils<0.22,>=0.20, but you have docutils 0.16 which is incompatible.\n",
            "gcsfs 2024.10.0 requires fsspec==2024.10.0, but you have fsspec 2024.12.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed awscli-1.37.2 botocore-1.36.2 colorama-0.4.6 docutils-0.16 rsa-4.7.2 s3transfer-0.11.1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "OdImX3_o9vxk"
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "mTeMHt8J9vN2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!aws configure"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NHsOmusc8DPk",
        "outputId": "b9feacba-0585-42f8-f5fd-4384345d9742"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "AWS Access Key ID [None]: AKIA5FCD6EEG3XKS6Q7R\n",
            "AWS Secret Access Key [None]: 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\n",
            "Default region name [None]: eu-north-1\n",
            "Default output format [None]: \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from s3fs import S3FileSystem\n",
        "s3 = S3FileSystem()"
      ],
      "metadata": {
        "id": "VJE1nQBl8-oU"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "for count,i in enumerate(consumer):\n",
        "    with s3.open(\"s3://kafka-stock-jay/stock_market_{}.json\".format(count), 'w') as file:\n",
        "        json.dump(i.value, file)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "5rqg9QMp9LSa",
        "outputId": "a9f73bcd-988c-40f8-d9d3-93458a03b385"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "ERROR:kafka.conn:<BrokerConnection node_id=0 host=16.171.32.64:9092 <connected> [IPv4 ('16.171.32.64', 9092)]>: socket disconnected\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.consumer.fetcher:Fetch to node 0 failed: KafkaConnectionError: socket disconnected\n",
            "ERROR:kafka.conn:<BrokerConnection node_id=0 host=16.171.32.64:9092 <connected> [IPv4 ('16.171.32.64', 9092)]>: socket disconnected\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.consumer.fetcher:Fetch to node 0 failed: KafkaConnectionError: socket disconnected\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n",
            "ERROR:kafka.conn:Connect attempt to <BrokerConnection node_id=0 host=16.171.32.64:9092 <connecting> [IPv4 ('16.171.32.64', 9092)]> returned error 111. Disconnecting.\n",
            "WARNING:kafka.client:Node 0 connection failed -- refreshing metadata\n"
          ]
        },
        {
          "output_type": "error",
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-31-c1f0a31b4003>\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mcount\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconsumer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0ms3\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"s3://kafka-stock-jay/stock_market_{}.json\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m         \u001b[0mjson\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdump\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1191\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_v1\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1192\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1193\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1194\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1195\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mnext_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36mnext_v2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1199\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_message_generator_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1200\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1201\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1202\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1203\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iterator\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m_message_generator_v2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1114\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_message_generator_v2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1115\u001b[0m         \u001b[0mtimeout_ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1000\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_consumer_timeout\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1116\u001b[0;31m         \u001b[0mrecord_map\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupdate_offsets\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1117\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mtp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrecords\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miteritems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord_map\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1118\u001b[0m             \u001b[0;31m# Generators are stateful, and it is possible that the tp / records\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36mpoll\u001b[0;34m(self, timeout_ms, max_records, update_offsets)\u001b[0m\n\u001b[1;32m    653\u001b[0m         \u001b[0mremaining\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtimeout_ms\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    654\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 655\u001b[0;31m             \u001b[0mrecords\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_poll_once\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mremaining\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_records\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupdate_offsets\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mupdate_offsets\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    656\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    657\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/consumer/group.py\u001b[0m in \u001b[0;36m_poll_once\u001b[0;34m(self, timeout_ms, max_records, update_offsets)\u001b[0m\n\u001b[1;32m    700\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    701\u001b[0m         \u001b[0mtimeout_ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coordinator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime_to_next_poll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_client\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout_ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m         \u001b[0;31m# after the long poll, we should check whether the group needs to rebalance\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m         \u001b[0;31m# prior to returning data so that the group can stabilize faster\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/client_async.py\u001b[0m in \u001b[0;36mpoll\u001b[0;34m(self, timeout_ms, future)\u001b[0m\n\u001b[1;32m    600\u001b[0m                     \u001b[0mtimeout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# avoid negative timeouts\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    601\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 602\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_poll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    603\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    604\u001b[0m             \u001b[0;31m# called without the lock to avoid deadlock potential\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/kafka/client_async.py\u001b[0m in \u001b[0;36m_poll\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    632\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    633\u001b[0m         \u001b[0mstart_select\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 634\u001b[0;31m         \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    635\u001b[0m         \u001b[0mend_select\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    636\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sensors\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.11/selectors.py\u001b[0m in \u001b[0;36mselect\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    466\u001b[0m             \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    467\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 468\u001b[0;31m                 \u001b[0mfd_event_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_ev\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    469\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mInterruptedError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    470\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ]
    }
  ]
}